### ISSUE TITLE
Unable to run Directly the "Pytorch Training" of the notebook as of missing Model (PATH = "/content/gdrive/My Drive/Yolov8/OWL-ViT_Models/with_CNN_on_Objects/model.pt") 

### ISSUE THREAD

**Issue by MadanMl** [melihsrn/OWL-ViT_with_enhanced_query_embedding_network#2]
Hi, Thanks for open-sourcing your notebook. I am trying to run directly the Pytorch training part in the notebook without running the Query Embedding Network part. These are steps I took.

**1 I added data to my drive and tried creating the data-loaders but saw that there was error in the following line due to the missing column__index_level_0__ : last_train_dataset= train_dataset.map(process_dataset,num_proc=4)
last_train_dataset = last_train_dataset.remove_columns(['image_x','image_y','objects_x','objects_y','__index_level_0__'])

last_test_dataset= test_dataset.map(process_dataset,num_proc=4)
last_test_dataset = last_test_dataset.remove_columns(['image_x','image_y','objects_x','objects_y','__index_level_0__'])

last_test_dataset_train_object= test_dataset_train_objects.map(process_dataset,num_proc=4)
last_test_dataset_train_object = last_test_dataset_train_object.remove_columns(['image_x','image_y','objects_x','objects_y','__index_level_0__'])

**2 Once you do the above part you are able to create data-loaders. **But I am stuck at the part where you have to load models from checkpoints**.  In the repository there is no "with_CNN_on_Objects/model.pt" available Here:
# Get Pre-trained CNN for Query Embedding Extraction
PATH = "/content/OWL-ViT_with_enhanced_query_embedding_network/query_embed_network/Object/model.pt"
checkpoint = torch.load(PATH)
query_embedding_model.load_state_dict(checkpoint['model_state_dict'])

model=OWL_ViT(_model,query_embedding_model)
model.to(device)
optimizer = torch.optim.AdamW(
    model.parameters(),
    lr=1e-6,
    weight_decay=1e-4,
)

# Get Pre-trained Model
PATH = "/content/gdrive/My Drive/Yolov8/OWL-ViT_Models/with_CNN_on_Objects/model.pt"
checkpoint = torch.load(PATH)
model.load_state_dict(checkpoint['model_state_dict'])
optimizer.load_state_dict(checkpoint['optimizer_state_dict'])
epoch = checkpoint['epoch']
epoch_losses = checkpoint['epoch_losses_df']
epoch_losses_test = checkpoint['epoch_losses_test_df']
epoch_losses_test_on_train_objects = checkpoint['epoch_losses_test_on_train_objects_df']

Can you help me to reproduce the results for the objects dataset using your script? thanks

**Comment by melihsrn** [comment]
Hello,
Thank you for running my code. For the first issue, you need to check the column names of your dataset after mapping so that you can see the column to be extracted. The name may have changed for your case and that's why you might be getting this error. For the second one, you probably did not change the path. I have already uploaded the models into repository. You need to get the repository to your local pc and use the path in which you store the model. 
I would be happy to help you for further issues but it may be hard to understand my code. I suggest you to dive into the code and try to understand the purposes of lines if you are willing to use the code for further purposes.
Best,

**Comment by MadanMl** [comment]
Hello Thank you for your reply.
I had actually used the dataset you provided on your repository therefore I couldn't understand why some labels were missing. For the part of the models I have added some questions in the following colab notebook (https://colab.research.google.com/drive/1cAtiNauDSzW3bNJMf-3iavvB3D_KlQlk?usp=sharing). In the notebook please can you check the section "Which model to add". As I understand, in your GitHub Repo there are only models available for query_embed_network (both for Digit and Object) but no model is available for ".......with_CNN_on_Objects/model.pt". 
Best,
I hope to understand your awesome work and hopefully create a notebook for fine-tuning OWL-VIT model just for object detection without any text queries but just based on YOLO formatted labels. I couldn't find a working script for this. This is my final goal. Thanks

**Comment by melihsrn** [comment]
Hello,
I understand your confusion. I have not added any pretrained overall model. The path you saw there was the path I used for temporary purposes while working on the project. You need to train the overall model and see the results yourself. About the dataset, yes, there might be some object folder with missing labels because they were first trials. You should use the ones with metadata csv file. 
I understand your project because I did the similar one as you see and I could not find any helpful resources, either. If you are not able to work on my notebook, I suggest you to go to the original OWL-ViT documentation and start over as that is what I did. 
Best,
